src/fu-engine.c

```diff
@@ -1611,16 +1611,18 @@ fu_engine_install (FuEngine *self,
 		if (caption != NULL) {
 			g_set_error (error,
 				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
+				     FWUPD_ERROR_NEEDS_USER,
 				     "Device %s needs to manually be put in update mode: %s",
 				     fu_device_get_name (device), caption);
 		} else {
 			g_set_error (error,
 				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
+				     FWUPD_ERROR_NEEDS_USER,
 				     "Device %s needs to manually be put in update mode",
 				     fu_device_get_name (device));
 		}
+		fu_device_set_update_state (device, FWUPD_UPDATE_STATE_FAILED_TRANSIENT);
+		fu_device_set_update_error (device, (*error)->message);
```

Comment:
This isn't good practice, as it's perfectly valid to set error to NULL in the caller. You need to do the error_local dance here.

Reply:
It's not though for this function.  Look 20 lines above and you'll see:
```	g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
```

Reply:
> g_return_val_if_fail (error == NULL || *error == NULL, FALSE);

If `error` is NULL then we don't return with failure. This is checking that either error is NULL, *or* *error is NULL and this unset.

Reply:
OK just pushed a fixup to test.  I'm going to be AFK for a while in meetings.  Feel free to fixup yourself if any other changes needed to get this into 1.3.3.